ui/static/ui/js/listing.js

```diff
@@ -417,156 +393,247 @@ define('listing',
+       * refresh has occurred.
+       */
+      updatePage: function (newPageNum) {
+        var queryMap = this.makeQueryMap();
+        queryMap.page = [newPageNum.toString()];
+
+        this.updateQuery(queryMap);
+        return this.refreshFromAPI();
+      },
 
-      // If search is executed update query parameter and refresh from API.
-      $("#search_button").click(function(e) {
-        e.preventDefault();
+      componentDidMount: function () {
+        CSRF.setupCSRF();
 
-        var search = $("#id_q").val();
-        updateSearch(search);
-      });
+        var thiz = this;
 
-      // Close exports panel.
-      $('.cd-panel-exports').on('click', function (event) {
-        if ($(event.target).is('.cd-panel-exports') ||
-          $(event.target).is('.cd-panel-close')) {
-          $('.cd-panel-exports').removeClass('is-visible');
+        $('[data-toggle=popover]').popover();
+        //Close panels on escape keypress
+        $(document).keyup(function (event) {
+          if (event.keyCode === 27) { // escape key maps to keycode `27`
+            closeLearningResourcePanel();
+            if ($('.cd-panel-2').hasClass('is-visible')) {
+              hideTaxonomyPanel();
+            }
+            if ($('.cd-panel-exports').hasClass('is-visible')) {
+              $('.cd-panel-exports').removeClass('is-visible');
+            }
+            if ($('.cd-panel-members').hasClass('is-visible')) {
+              $('.cd-panel-members').removeClass('is-visible');
+            }
+            event.preventDefault();
+          }
+        });
+
+        //close the lateral panel
+        $('.cd-panel').on('click', function (event) {
+          if ($(event.target).is('.cd-panel') ||
+            $(event.target).is('.cd-panel-close')) {
+            closeLearningResourcePanel();
+            event.preventDefault();
+          }
+        });
+
+        //open the lateral panel
+        $('.btn-taxonomies').on('click', function (event) {
           event.preventDefault();
-        }
-      });
-
-      //open the lateral panel for members
-      $('.btn-members').on('click', function (event) {
-        event.preventDefault();
-        //remove any alert
-        $('#members-alert').empty();
-        //reset the form values
-        resetUserGroupForm();
-        //make panel visible
-        $('.cd-panel-members').addClass('is-visible');
-        //retrieve all the members
-        showUpdateAllMembers();
-        //
-      });
-      //close the lateral panel for members
-      $('.cd-panel-members').on('click', function (event) {
-        if ($(event.target).is('.cd-panel-members') ||
-          $(event.target).is('.cd-panel-close')) {
-          $('.cd-panel-members').removeClass('is-visible');
+          thiz.loadManageTaxonomies();
+          $('.cd-panel-2').addClass('is-visible');
+        });
+
+        //close the lateral panel
+        $('.cd-panel-2').on('click', function (event) {
```

Comment:
Does React have a thing like Backbone where you can provide a mapping of event+selector to function to call? As it stands, this method is really long and that might help clean it up. If not, perhaps something to consider writing in the future?